runtime/vm/createramclass.cpp

```diff
@@ -2294,7 +2294,7 @@ internalCreateRAMClassDone(J9VMThread *vmThread, J9ClassLoader *classLoader, J9C
 		}
 
 #if defined(J9VM_OPT_VALHALLA_VALUE_TYPES)
-		if (J9ROMCLASS_HAS_IDENTITY(romClass)) {
+		if (!J9ROMCLASS_IS_VALUE(romClass)) {
 			classFlags |= J9ClassHasIdentity;
 		}
```

Comment:
Shouldn't `J9ClassHasIdentity` be set (unconditionally) if `!defined(J9VM_OPT_VALHALLA_VALUE_TYPES)`?
```c
		if (!J9ROMCLASS_IS_VALUE(romClass)) {
			classFlags |= J9ClassHasIdentity;
		}
#if defined(J9VM_OPT_VALHALLA_VALUE_TYPES)
```

Reply:
I'm not sure this flag is actually needed anymore, in an old spec it was previously used to distinguish non-value types with certain characteristics. I've started some cleaning up of its uses in https://github.com/eclipse-openj9/openj9/pull/19716 and plan to have another look

Reply:
It hasn't been the case in the past that it was set for `!defined(J9VM_OPT_VALHALLA_VALUE_TYPES)`

Reply:
Perhaps it wasn't set before, but I think that was an oversight. There places in the JIT that check that flag to distinguish regular classes from value types. Those will get the wrong answer unless the flag is set even if `!defined(J9VM_OPT_VALHALLA_VALUE_TYPES)`. I suggest line 3446 (for arrays) should also be unconditional.